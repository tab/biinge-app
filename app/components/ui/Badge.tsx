import React from "react"
import { View, Text } from "react-native"

import { badgeStyles } from "styles"

type Props = {
  children: React.ReactNode
}

const BadgeComponent = ({ children }: Props) => {
  return (
    <View style={badgeStyles.container}>
      <Text style={badgeStyles.text}>{children}</Text>
    </View>
  )
}

export default BadgeComponent
