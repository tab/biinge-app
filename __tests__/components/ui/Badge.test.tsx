import React from "react"
import "react-native"

import { it } from "@jest/globals"
import renderer from "react-test-renderer"

import Badge from "components/ui/Badge"

describe("Badge", () => {
  const title = "Badge title"

  it("renders correctly", () => {
    const tree = renderer.create(<Badge>{title}</Badge>)
    expect(tree).toMatchSnapshot()
  })
})
