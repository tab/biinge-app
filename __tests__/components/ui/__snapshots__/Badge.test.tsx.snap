// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(25, 25, 25, 0.25)",
      "borderRadius": 50,
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingBottom": 2,
      "paddingHorizontal": 10,
      "paddingVertical": 0,
      "position": "absolute",
      "right": 18,
      "top": 22,
    }
  }
>
  <Text
    style={
      {
        "color": "#ffffff",
        "fontSize": 12,
        "fontWeight": "400",
        "lineHeight": 21,
      }
    }
  >
    Badge title
  </Text>
</View>
`;
